src/layer/AtmosphereLayer.js

```diff
@@ -0,0 +1,414 @@
+                    this._lightLocation = value;
+                }
+            },
+
+            /**
+             * Url for the night texture.
+             * @memberof AtmosphereLayer.prototype
+             * @type {URL}
+             */
+            nightImageSource: {
+                get: function () {
+                    return this._nightImageSource;
+                },
+                set: function (value) {
+                    this._nightImageSource = value;
+                }
+            },
+
+            /**
+             * The number of longitudinal points in the grid.
+             * @memberof AtmosphereLayer.prototype
+             * @type {Number}
+             */
+            skyWidth: {
+                get: function () {
+                    return this._skyWidth;
+                },
+                set: function (value) {
+                    this._skyWidth = value;
+                }
+            },
+
+            /**
+             * The number of latitudinal points in the grid.
+             * @memberof AtmosphereLayer.prototype
+             * @type {Number}
+             */
+            skyHeight: {
+                get: function () {
+                    return this._skyHeight;
+                },
+                set: function (value) {
+                    this._skyHeight = value;
+                }
+            },
+
+            /**
+             * The grid of sky Cartesian points.
+             * @memberof AtmosphereLayer.prototype
+             * @type {Float32Array}
+             */
+            skyPoints: {
+                get: function () {
+                    return this._skyPoints;
+                },
+                set: function (value) {
+                    this._skyPoints = value;
+                }
+            },
+
+            /**
+             * The triangle strip sky indices.
+             * @memberof AtmosphereLayer.prototype
+             * @type {Number[]}
+             */
+            skyTriStrip: {
+                get: function () {
+                    return this._skyTriStrip;
+                },
+                set: function (value) {
+                    this._skyTriStrip = value;
+                }
+            }
+        });
+
+        // Documented in superclass.
+        AtmosphereLayer.prototype.doRender = function (dc) {
+
+            if (dc.globe.is2D()) {
+                return;
+            }
+
+            this.determineLightDirection(dc);
+            this.drawSky(dc);
+            this.drawGround(dc);
+        };
+
+        // Internal. Intentionally not documented.
+        AtmosphereLayer.prototype.applySkyVertices = function (dc) {
+
+            var gl = dc.currentGlContext,
+                program = dc.currentProgram,
+                skyData = this.skyData,
+                vboId;
+
+            if (!skyData.verticesVboCacheKey) {
+                skyData.verticesVboCacheKey = dc.gpuResourceCache.generateCacheKey();
+            }
+
```

Comment:
The adaptation of sky vertex buffer objects from WWAndroid to WebWW is generally quite good. However, I don't see a need for the refreshVertexBuffer and refreshIndices buffer properties. It would be better if the buffer were created and its data specified in one step.